core/modules/replica/ReplicateJob.h

```diff
@@ -109,7 +109,7 @@ class ReplicateJob
     ReplicateJob(ReplicateJob const&) = delete;
     ReplicateJob& operator=(ReplicateJob const&) = delete;
 
-    ~ReplicateJob() = default;
+    ~ReplicateJob() final = default;
```

Comment:
I'm confused; the coliru compiler (g++ -std=c++11 -O2 -Wall -pedantic -pthread main.cpp && ./a.out) says it's an error to declare a member function `final` if it's not declared `virtual`. I reread the [cppreference page about 'final'](https://en.cppreference.com/w/cpp/language/final), it says
> When used in a virtual function declaration or definition, final ensures that the function is virtual and specifies that it may not be overridden by derived classes. The program is ill-formed (a compile-time error is generated) otherwise.
which I originally took to mean it makes the function virtual, but after reading it closer I think it means it _requires_ the function to be declared virtual by the programmer. But then, I wonder why our compiler is not complaining?

Also, I found one thread suggesting that since the effect of declaring the destructor `final` makes the class final then the more explicit thing to do is to declare the class itself final instead (and don't declare the dtor final). Did you consider this already, do you have any thoughts about it?

Reply:
Goof suggestion indeed to declare the whole class a `final`:
```
class Subclass final : public Base {
   ...
```

Reply:
Goof(y) suggestion indeed 😆